discord/state.py

```diff
@@ -1612,6 +1612,54 @@ def parse_entitlement_delete(self, data: gw.EntitlementDeleteEvent) -> None:
         entitlement = Entitlement(data=data, state=self)
         self.dispatch('entitlement_delete', entitlement)
 
+    def parse_message_poll_vote_add(self, data: gw.PollVoteActionEvent) -> None:
+        # NOTE: though the data contains an ``answer_id`` it means nothing.
+        # We shouldn't rely on them, as their ID it is just their position
+        # in the poll UI.
+
+        raw = RawPollVoteActionEvent(data)
+
+        self.dispatch('raw_poll_vote_add', raw)
+
+        message = self._get_message(raw.message_id)
+        guild = self._get_guild(raw.guild_id)
+
+        if guild:
+            user = guild.get_member(raw.user_id)
+        else:
+            user = self.get_user(raw.user_id)
+
+        if message and user:
+            if not message.poll:
+                _log.warning('Cannot dispatch "poll_vote_add" event because the cached message does not contain a poll. This is a caching error.')
```

Comment:
```suggestion
                _log.warning(
                    'POLL_VOTE_ADD referencing message with ID: %s without a poll. Discarding.',
                    raw.message_id,
                )
```

I'd probably reword this, same for the one for remove, at a mimimum including the message ID would be desirable.